### PR TITLE
fix style in app as per rubocop

### PR DESCRIPTION
**Pull request by georgestam**
@vemv I have added `rubocop` and fixed almost everything. Then, I fixed a number of things also but I think that the following we would leave it for now... as it is too much work (not sure about the last one):

screen shot 2017-03-26 at 19 58 47

### CODE REVIEW

File: spec/factories/imparted_hour_factory.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
     journey
     default_minutes = Random.rand(1..120)
     minutes { default_minutes }
-    date { Date.today }
+    date { Time.zone.today }
```
Comment (vemv) [inline spec/factories/imparted_hour_factory.rb]: use `Time.current.to_date` instead. Same result, but you should be using `Time.current` for almost everything.

The reason is that 1 thing is easier to memorize than N things

File: app/controllers/api/v1/base_controller.rb [hunk 2]
```diff
@@ -1,31 +0,0 @@
-class Api::V1::BaseController < ActionController::Base
```
Comment (vemv) [inline app/controllers/api/v1/base_controller.rb]: did rubocop detect an unused controller?

File: app/controllers/profiles_controller.rb [hunk 3]
```diff
@@ -33,12 +32,12 @@ def dashboard
     @user = current_user
     authorize @user
     @teacher = false
+    car = @user.cars.first ? @user.cars.first : nil
```
Comment (vemv) [inline app/controllers/profiles_controller.rb]: leave it as `car = @user.cars.first`

File: app/controllers/profiles_controller.rb [hunk 4]
```diff
@@ -116,7 +114,7 @@ def find_car
   end
 
   def user_params
-    params.require(:user).permit(:photo, :id_document, :first_name, :last_name, :email, :phone_number, :description, :gender, :facebook_URL, :linkedin_URL, :address, :date_of_birth)
+    params.require(:user).permit(:photo, :id_document, :first_name, :last_name, :email, :phone_number, :description, :gender, :facebook_url, :linkedin_url, :address, :date_of_birth)
   end
```
Comment (vemv) [inline app/controllers/profiles_controller.rb]: not sure if params are case-insensitive. I recommend downcase as rubocops does, but in that case ensure that this change is applied everywhere

File: app/views/cars/_card.html.erb [hunk 5]
```diff
@@ -93,7 +93,7 @@
     </li>
     <li>
       <p>
-        <% if car.user.facebook_URL != "" %>
+        <% if car.user.facebook_url != "" %>
```
Comment (vemv) [inline app/views/cars/_card.html.erb]: careful with this case change, as said

File: config/database.yml [hunk 6]
```diff
@@ -67,15 +67,15 @@ test:
...
 # available as an environment variable. For example:
 #
-#   DATABASE_URL="postgres://myuser:mypass@localhost/somedatabase"
+#   DATABASE_url="postgres://myuser:mypass@localhost/somedatabase"
 #
 # You can use this database configuration with:
 #
 #   production:
-#     url: <%= ENV['DATABASE_URL'] %>
+#     url: <%= ENV['DATABASE_url'] %>
```
Comment (vemv) [inline config/database.yml]: I'd just remove the rubocop rule which is rewriting "URL". Seems too dangerous.

Use `bundle exec rubocop -d` to determine which rule is being applied, then add it to rubocop.yml

File: config/environments/development.rb [hunk 7]
```diff
@@ -15,7 +15,7 @@
   config.consider_all_requests_local = true
 
   # Enable/disable caching. By default caching is disabled.
-  if Rails.root.join('tmp/caching-dev.txt').exist?
+  if Rails.root.join('path', 'to').exist?
```
Comment (vemv) [inline config/environments/development.rb]: looks strange.

File: config/initializers/devise.rb [hunk 8]
```diff
@@ -249,8 +249,8 @@
   # Add a new OmniAuth provider. Check the wiki for more information on setting
   # up on your models and hooks.
   # config.omniauth :github, 'APP_ID', 'APP_SECRET', scope: 'user,public_repo'
-  Devise.setup do |config|
-    config.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
+  Devise.setup do |con|
+    con.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
```
Comment (vemv) [inline config/initializers/devise.rb]: bizarre variable name change! was it automatic? if so, avoid it with rubocop.yml (as said above)

File: app/controllers/api/v1/base_controller.rb (hunk 2 above)
Comment (georgestam) [inline app/controllers/api/v1/base_controller.rb]: Rubocop detected a number of offences, as it is not used at all, I decided to remove it. 
screen shot 2017-03-27 at 09 19 57

Comment (vemv) [inline app/controllers/api/v1/base_controller.rb]: that kind of action deserves its own commit :)

File: config/database.yml (hunk 6 above)
Comment (georgestam) [inline config/database.yml]: The whole things about `URL` upcase/downcase was my fault (I did a 'Replace All' destruction). Now I have only changed the instance variables that `robocop complained, which were less than 10.

File: config/environments/development.rb (hunk 7 above)
Comment (georgestam) [inline config/environments/development.rb]: screen shot 2017-03-27 at 10 18 10

Comment (vemv) [inline config/environments/development.rb]: it's complaining about the slash. Prefer `.join` to `/`. "path" and "to" are examples.

File: config/initializers/devise.rb (hunk 8 above)
Comment (georgestam) [inline config/initializers/devise.rb]: Rubocop complained about the use of the use of word `config`, that's why I changed to `con`. I changed now to `var`? (by the way, the feature with facebook login it is disabled for now)

screen shot 2017-03-27 at 10 24 44

Comment (vemv) [inline config/initializers/devise.rb]: You're fixing the symptom rather than the cause. Nesting `Devise.setup` blocks one inside the other makes no sense. Extract the content of the inner one, and put it in the outer one.

**Comment by georgestam** [comment]
@vemv I have implemented the changes as follows:
- Commit bdd9539 shows all the changes that `rubocop` fixed automatically
- Commit 4efa3ec shows the changes that I fixed manually. There are still 6 offences as per the first snapshot above. 

(Commit 	eb834d4 fix some issues with pundit that I did not identify earlier... due to the lack of testing!)

File: config/environments/development.rb (hunk 7 above)
Comment (vemv) [inline config/environments/development.rb]: not fixed

File: config/initializers/devise.rb [hunk 9]
```diff
@@ -249,8 +249,8 @@
   # Add a new OmniAuth provider. Check the wiki for more information on setting
   # up on your models and hooks.
   # config.omniauth :github, 'APP_ID', 'APP_SECRET', scope: 'user,public_repo'
-  Devise.setup do |config|
-    config.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
+  Devise.setup do |var|
+    var.omniauth :facebook, ENV["FB_ID"], ENV["FB_SECRET"],
```
Comment (vemv) [inline config/initializers/devise.rb]: not fixed, avoid nesting Devise.setup inside Devise.setup

File: spec/factories/imparted_hour_factory.rb [hunk 10]
```diff
@@ -5,7 +5,7 @@
     journey
     default_minutes = Random.rand(1..120)
     minutes { default_minutes }
-    date { Date.today }
+    date { Time.current }
```
Comment (vemv) [inline spec/factories/imparted_hour_factory.rb]: `Time.current.to_date`? (I assume we want a Date object)

File: app/controllers/profiles_controller.rb [hunk 11]
```diff
@@ -57,24 +56,20 @@ def dashboard
       calculate_avg_rating
 
       @passengers = Passenger.where(journey_id: Journey.where(car: @car))
-
-      end
```
Comment (vemv) [inline app/controllers/profiles_controller.rb]: This moved `end` (it's below now) might change meaning. Or maybe git generated a strange diff (happens sometimes)?

File: config/environments/development.rb (hunk 7 above)
Comment (georgestam) [inline config/environments/development.rb]: `Rails.root.join('tmp/caching-dev.txt').exist`is a default comand from rails 5. I changed it now but I am unsure it is correct (http://stackoverflow.com/questions/3724487/rails-root-directory-path)

File: config/initializers/devise.rb (hunk 9 above)
Comment (georgestam) [inline config/initializers/devise.rb]: Not sure what I have to do here. I would need to study properly how omniauth works (I was not the one who wrote this code). I would propose to remove everything related with omniauth and implement this later in a new commit.

File: app/controllers/profiles_controller.rb (hunk 11 above)
Comment (georgestam) [inline app/controllers/profiles_controller.rb]: Yes, it is correct. The code worked also before but now is better, as a part of the code is not run now if the teacher is the current user.

File: app/controllers/api/v1/base_controller.rb (hunk 2 above)
Comment (georgestam) [inline app/controllers/api/v1/base_controller.rb]: Ok, I put back the code and will remove it in another commit.

File: config/initializers/devise.rb (hunk 9 above)
Comment (vemv) [inline config/initializers/devise.rb]: > avoid nesting Devise.setup inside Devise.setup
do this, nothing else needed

File: config/environments/development.rb (hunk 7 above)
Comment (vemv) [inline config/environments/development.rb]: it's correct now. Anyway you could have tried both pieces in `rails c` and checking that they are in fact equivalent

**Comment by georgestam** [comment]
@vemv I have implemented the small change

File: db/seeds.rb [hunk 12]
```diff
@@ -165,27 +156,26 @@
...
-    date_of_birth: Time.new(2002, 10, 31),
+    date_of_birth: Time.new.utc(2002, 10, 31),
     address: "Plaça de la Paeria, 1, 25007 Lleida, Spain",
     admin: true
     })
-  admin.photo = open("http://res.cloudinary.com/georgestam/image/upload/v1476103177/dtwp5e5fj2vqiv0mkacn.jpg")
-  admin.save
-
+admin.photo = open("http://res.cloudinary.com/georgestam/image/upload/v1476103177/dtwp5e5fj2vqiv0mkacn.jpg")
+admin.save
```
Comment (vemv) [inline db/seeds.rb]: .save!
